src/DynamoCoreWpf/ViewModels/Core/NoteViewModel.cs

```diff
@@ -219,5 +263,144 @@ private bool CanAddToGroup(object parameters)
             }
             return false;
         }
+
+        private void PinToNode(object parameters)
+        {
+
+            var nodeToPin = DynamoSelection.Instance.Selection
+                .OfType<NodeModel>()
+                .FirstOrDefault();
+
+            if (nodeToPin == null)
+            {
+                return;
+            }
+
+            Model.PinnedNode = nodeToPin;
+
+            MoveNoteAbovePinnedNode();
+
+            SubscribeToPinnedNode();
+
+        }
+
+        private bool CanPinToNode(object parameters)
+        {
+            // Go over all elements selected and get the first NodeModel,
+            // If there is no nodeModel or there is already a node pinned
+            // Note cannot be pinned to any node
+
+            var nodeToPin = DynamoSelection.Instance.Selection
+                    .OfType<NodeModel>()
+                    .FirstOrDefault();
+
+            if (nodeToPin == null || Model.PinnedNode != null)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private void UnpinFromNode(object parameters)
+        {
+            UnsuscribeFromPinnedNode();
+            Model.PinnedNode = null;
+        }
+
+        private void SubscribeToPinnedNode()
+        {
+            // Subscribe to PinnedNode (model and viewmodel) Property_Changed
+            // so that note moves above and behind node every time
+            // NodeModel.State, NodeModel.Position, or NodeViewModel.ZIndex change
+            Model.PinnedNode.PropertyChanged += PinnedNodeModel_PropertyChanged;
+            PinnedNode.PropertyChanged += PinnedNodeViewModel_PropertyChanged;
+
+            // Subscribe to pinnedNode.RequestSelection (fires before node is selected)
+            // so that this note is added to the selection
+            //PinnedNode.RequestsSelection += PinnedNodeViewModel_RequestsSelection;
+            PinnedNode.Selected += PinnedNodeViewModel_RequestsSelection;
```

Comment:
rename `RequestsSelection` should probably be something like `OnPinnedNodeSelected`